### PR TITLE
gh-2988 App crashes when using QR code to connect key via Walletconnect

### PR DESCRIPTION
**Pull request by biafra23**
Handles #3017  

Changes proposed in this pull request:
- Wallet can be nil for importKey

### CODE REVIEW

File: Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift [hunk 1]
```diff
@@ -44,14 +44,13 @@ class WalletConnectKeyFlow: AddKeyFlow {
 
     override func doImport() -> Bool {
         guard let connection = parameters?.connection,
-                let wallet = parameters?.wallet,
```
Comment (MouazAlzahabi) [inline Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: For data integrity I think we need different validation rules, so if we don't have wallet selected we need to make sure the url is valid and can be used for creating connection is that fully covered by the other checks ?

Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: Maybe I am wrong but it looks to me this worked in 3.18.1
How do I check the URL is valid? Do we have a test for that?

Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: I can try and find out what happens with a slightly broken URL

Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: This is what happens on the side that creates the QR-code. Do we really need to check we created a valid URL?

Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: Is it safe to assume that we need either a non-nil cdConnection or a non-nil wallet [here](https://github.com/safe-global/safe-ios/blob/0b05a77e4a0d722bf78a7a0cbfdcaa9f3e3ab054/Multisig/Logic/Models/KeyInfo.swift#L211) :
```
        if let cdConnection = CDWCConnection.connection(by: connection.connectionURL.absoluteString) {
            item.addToConnections(cdConnection)
        }

        if let wallet = wallet, let cdRegistryEntry = CDWCAppRegistryEntry.entry(by: wallet.id) {
            item.wallet = cdRegistryEntry
        }
```

Comment (biafra23) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/WalletConnectKeyFlow.swift:47]: I don't think it is possible to get here (`doImport()`) without a connection. The crashing app showed a QR code, that is scanned by a wallet. The crash happens, after the other wallet connects to the crashing app when we enter the name for the imported key.

**Pull request closed without merging**